File: terre10.py
import sys

def est_nombre_premier(nombre):
    if nombre <= 1:
        return False
    if nombre <= 3:
        return True

    if nombre % 2 == 0 or nombre % 3 == 0:
        return False

    i = 5
    while i * i <= nombre:
        if nombre % i == 0 or nombre % (i + 2) == 0:
            return False
        i += 6

    return True

if len(sys.argv) != 2:
    print("erreur.")
else:
    nombre = sys.argv[1]

    if nombre.isdigit():
        nombre = int(sys.argv[1])

        if est_nombre_premier(nombre):
            print(f"{nombre} est un nombre premier.")
        else:
            print(f"{nombre} n'est pas un nombre premier.")

    else:
        print("erreur.")


    

File: terre10.js
const args = process.argv.slice(2)

if (args.length !== 1) {
  console.error("Le programme a besoin d'un unique argument pour fonctionner.")
  process.exit()
}

if (isNaN(args[0])) {
  console.error("L'argument doit être un nombre entier.")
  process.exit()
}

const number = +args[0]

if (number < 0) {
  console.error("L'argument doit être un entier positif.")
  process.exit()
}

let isPrimeNumber = true

const exceptionNumbers = [0, 1]

if (exceptionNumbers.includes(number)) isPrimeNumber = false

for (let i = 2; i < number; i++) {
  if (number % i === 0) isPrimeNumber = false
}

if (isPrimeNumber) {
  console.log(`Oui, ${number} est un nombre premier.`)
  process.exit()
}
console.log(`Non, ${number} n'est pas un nombre premier.`)
